src/sql-parser/src/ast/defs/name.rs

```diff
@@ -94,6 +94,12 @@ impl_display!(Ident);
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct ObjectName(pub Vec<Ident>);
 
+impl From<&str> for ObjectName {
+    fn from(n: &str) -> ObjectName {
+        ObjectName(vec![Ident::new(n)])
+    }
+}
```

Comment:
This is a little too magical for my taste. In general I'm sort of leery of adding `From` implementations unless there is a very obvious correspondence between two types, and it's the implementation that you want 95%+ of the time. What do you think about a method like `ObjectName::unqualified` or somesuch instead? I find something like

```
ObjectName::unqualified("foo")
```

to more obviously be about a) creating an object name, and b) not trying to do any fancy parsing to separate components than

```
"foo".into()
```

Reply:
Given the prevalence of casting `&str` to `ObjectName`, felt like it passed this test, but can appreciate it being too magical. Will fix this up with the rest of your comments.